app/updatemanager.go

```diff
@@ -326,17 +342,19 @@ func (u *UpdateControlMap) String() string {
 }
 
 type ControlMapPool struct {
-	Pool  []*UpdateControlMap
-	mutex sync.Mutex
-	store store.Store
+	Pool    []*UpdateControlMap
+	mutex   sync.Mutex
+	store   store.Store
+	Updates chan struct{} // Announces all updates to the maps
 }
 
 // loadTimeout is how far in the future to set the map expiry when loading from
 // the store.
 func NewControlMap(store store.Store, loadTimeout int) *ControlMapPool {
 	pool := &ControlMapPool{
-		Pool:  []*UpdateControlMap{},
-		store: store,
+		Pool:    []*UpdateControlMap{},
+		store:   store,
+		Updates: make(chan struct{}, 1),
```

Comment:
Give the channel a 1-size buffer, and simply ignore more notifications filling up. Hence the non-blocking:
`select{
<- write
}`
calls

Reply:
Makes sense!